### DISCUSSION TITLE
Customization of the guest operating system is not supported due to the given reason.

### DISCUSSION THREAD

**Discussion by arkadicolson** [vmware/packer-examples-for-vsphere#788] [Help] [answered]
Hi

My setup is building a Debian 11 vm which works great and the export to the content library in vsphere as well. However when I try to deploy a vm from the ovf in the content library with terraform the vm is known in vcenter as a "Debian 11" type vm. So the deploy failes with message: error sending customization spec: Customization of the guest operating system is not supported due to the given reason:
The config however contains vm_guest_os_type = "ubuntu64Guest".
Any idea why the guest_os_type is not "ubuntu64Guest" when exported to the library?

Thx!

**Comment by tenthirtyam** [comment, marked as answer]
When it comes from the content library vSphere actually detects the OS as Debian 11.

**Comment by arkadicolson** [comment]
That's not completely true. when building a vm manually, exporting as ovf, importing into the content library it is detected correctly when deploying with terraform.
I only cannot pinpoint the problem yet
